### PR TITLE
API provides journalist to sign in to be able to create article

### PR DESCRIPTION
**Pull request by pgaunitz**
# [PT Story](https://www.pivotaltracker.com/story/show/171984574)
```
As an journalist,
In order to create an article,
I would like the site to have a unique way for an journalist to sign in
```

## Changes proposed in this pull request:
* Adds unit test for User model, and request spec for registration and login
* Installs and setups Devise Token Auth
* Ads roles to user model as enum
## What I have learned working on this feature:
* Modifying User model from Devise Token Auth with new DB column, and configuring permitted params 
* Refreshed knowledge about implementing and testing Devise Token Auth
* Learned more about authorization and restrictions
## Packages/Gems added
-  Devise Token Auth

### CODE REVIEW

File: config/initializers/devise_token_auth.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  #                        :'client' => 'client',
+  #                        :'expiry' => 'expiry',
+  #                        :'uid' => 'uid',
+  #                        :'token-type' => 'token-type' }
+
+  # By default, only Bearer Token authentication is implemented out of the box.
+  # If, however, you wish to integrate with legacy Devise authentication, you can
+  # do so by enabling this flag. NOTE: This feature is highly experimental!
+  # config.enable_standard_devise_support = false
+end
```
Comment (emtalen) [inline config/initializers/devise_token_auth.rb]: clean up comments

File: spec/requests/api/v1/articles_can_be_created_spec.rb [hunk 2]
```diff
@@ -1,124 +1,168 @@
...
       post '/api/v1/articles',
-      params: {
-        article: {
-          title: 'No more room in space',
-          snippet: 'Its all gone, sorry',
-          content: 'Govenor says this aint good',
-          category: 'tech'
-        }
-      }
+           params: {
```
Comment (emtalen) [inline spec/requests/api/v1/articles_can_be_created_spec.rb:11]: formatting is off

**Review by emtalen — COMMENTED** [review]
Just formatting and cleaning up some comments, good job! 💯

File: spec/requests/api/v1/articles_can_be_created_spec.rb [hunk 3]
```diff
@@ -1,124 +1,168 @@
...
+             }
+           },
+           headers: headers
     end
+
+    it 'returns 401 status' do
+      expect(response.status).to eq 401
+    end
+
+  it 'returns stuff' do
```
Comment (janko101) [inline spec/requests/api/v1/articles_can_be_created_spec.rb]: Just to put these little it block guys inline

File: spec/requests/api/v1/registration_spec.rb [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+           },
+           headers: headers
+    end
+
+    it "returns a 200 response status" do
+      expect(response_json["status"]).to eq "success"
+    end
+
+      it "returns a success message" do
+      expect(response_json["status"]).to eq "success"
```
Comment (janko101) [inline spec/requests/api/v1/registration_spec.rb:20]: Same..just to format

File: spec/requests/api/v1/sessions_spec.rb [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    end
+
+    it 'returns 401 response status' do
+      expect(response).to have_http_status 401
+    end
+
+    it 'returns error message' do
+      expect(response_json['errors']).to eq [
+           'Invalid login credentials. Please try again.'
+         ]
```
Comment (janko101) [inline spec/requests/api/v1/sessions_spec.rb:46]: Maybe to put ['errors'][0],so it comes the string

File: spec/requests/api/v1/sessions_spec.rb [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
```
Comment (janko101) [inline spec/requests/api/v1/sessions_spec.rb:64]: Same thing...try to get into the array

**Pull request merged**